core/common/pom.xml

```diff
@@ -31,6 +31,10 @@
 
   <dependencies>
     <!-- External dependencies -->
+    <dependency>
+      <groupId>com.amazonaws</groupId>
+      <artifactId>aws-java-sdk-core</artifactId>
```

Comment:
That's 
in UpdateCheck
```
  private static void addEC2Info(List<String> info) {
    boolean isEC2 = false;
    String productCode = EnvironmentUtils.getEC2ProductCode(); <-- here
    if (!productCode.isEmpty()) {
      info.add(String.format(PRODUCT_CODE_FORMAT, productCode));
      isEC2 = true;
    }

    String userData = "";
    try {
      userData = EC2MetadataUtils.getUserData(); <-- here
```
in EnvironmentalUtils
```
import com.amazonaws.util.EC2MetadataUtils;
  /**
   * Gets the EC2 product code if any.
   *
   * @return the first product code if any, an empty string otherwise
   */
  public static String getEC2ProductCode() {
    try {
      List<String> productCodes = EC2MetadataUtils.getProductCodes();
      if (productCodes == null || productCodes.size() < 1) {
        return "";
      }
      return productCodes.get(0);
    } catch (Throwable t) {
      // Exceptions are expected if this instance is not EC2 instance
      // or this EC2 does not have product code
      return "";
    }
  }

```

Reply:
@beinan in 3.0 we can make it a separate component? so that it can contain more info?